src/render/draw_circle.js

```diff
@@ -12,6 +12,7 @@ module.exports = drawCircles;
 
 function drawCircles(painter: Painter, sourceCache: SourceCache, layer: CircleStyleLayer, coords: Array<TileCoord>) {
     if (painter.renderPass !== 'translucent') return;
+    if (layer.paint['circle-opacity'] === 0 && layer.paint['circle-stroke-opacity'] === 0) return;
```

Comment:
With an eye towards https://github.com/mapbox/mapbox-gl-js/issues/3044, we should use `layer.isPaintValueFeatureConstant()` and `layer.getPaintValue()` instead of directly accessing `layer.paint`